master/internal/rm/agentrm/agent_resource_manager.go

```diff
@@ -389,12 +389,16 @@ func (*ResourceManager) IsReattachableOnlyAfterStarted() bool {
 
 // MoveJob implements rm.ResourceManager.
 func (a *ResourceManager) MoveJob(msg sproto.MoveJob) error {
-	for _, pool := range a.pools {
-		err := pool.MoveJob(msg)
-		if err != nil {
-			a.syslog.WithError(err).Errorf("failed to move job %s", msg.ID)
-			return err
-		}
+	pool, err := a.poolByName(msg.ResourcePool)
+	if err != nil {
+		a.syslog.WithError(err).Warnf("move job found no resource pool with name %s",
```

Comment:
just wrap the error and return it, don't bother with a log.

when i refactored this recently, I just left the log from the call there because it was already there pre-refactor. really we should follow https://github.com/uber-go/guide/blob/master/style.md#handle-errors-once so we don't get lots of extra logs.

apply this all over this change.